pkg/rule/rule.go

```diff
@@ -87,6 +88,60 @@ func (r *RuleTarget) Exists(c *nftables.Conn, ruleData RuleData) (bool, error) {
 	return true, nil
 }
 
+func (r *RuleTarget) Update(c *nftables.Conn, rules []RuleData) (bool, error) {
+	var modified bool
+	existingRules, err := c.GetRules(r.Table, r.Chain)
+	if err != nil {
+		return false, fmt.Errorf("error getting existing rules for update: %v", err)
+	}
+
+	add, remove := genRuleDelta(existingRules, rules)
+
+	for _, rule := range remove {
+		wasDeleted, err := r.Delete(c, rule)
+		if err != nil {
+			return false, err
+		}
+		if wasDeleted {
+			modified = true
+		}
+
+	}
+
+	for _, rule := range add {
+		wasAdded, err := r.Add(c, rule)
+		if err != nil {
+			return false, err
+		}
+		if wasAdded {
+			modified = true
+		}
+	}
+
+	return modified, nil
+}
+
+func genRuleDelta(existingRules []*nftables.Rule, newRules []RuleData) (add []RuleData, remove []RuleData) {
+	existingRuleMap := make(map[string]*nftables.Rule)
```

Comment:
Is it possible (or make sense) to use lists instead of a map, `bytes.Equal` and avoid the `string(UserData)` bit? This implementation is very obvious which is great but I am not a huge fan of the byte-string conversion requirement to use a map. If a list approach makes things more confusing or significantly more loop-y we will stick with the map.